pkg/cluster/util.go

```diff
@@ -227,13 +227,19 @@ func (c *Cluster) logServiceChanges(role PostgresRole, old, new *v1.Service, isU
 	}
 }
 
-func (c *Cluster) getPostgresContainer(podSpec *v1.PodSpec) v1.Container {
+func getPostgresContainer(podSpec *v1.PodSpec) v1.Container {
 	var pgContainer v1.Container
+
 	for _, container := range podSpec.Containers {
 		if container.Name == constants.PostgresContainerName {
 			pgContainer = container
 		}
 	}
+
+	// if no postgres container was found, take the first one in the podSpec
```

Comment:
that sounds very suspicious. Either it is simply an error or a PG operator deployment with non-standard container names (for example a very old one from a couple of years ago). Silently choosing the very first container is not the very best option in both cases.
That is still likely to work because of how we used container names before, but at least that case need to be acknowledged as a warning.